metadata-transformer/src/main/java/no/unit/nva/metadata/service/MetadataService.java

```diff
@@ -130,43 +124,46 @@ private Model getMetadata(URI uri) throws ExtractionException, IOException, URIS
         Model model = new TreeModel();
         for (Statement statement : statements) {
-            if (isDoi(statement)) {
-                Optional<Statement> doi = toBiboDoi(statement);
-                doi.ifPresent(model::add);
-            } else if (isSindiceDcOrDcTerms(statement)) {
-                model.add(toDctermsNamespace(statement));
-            } else if (isHighwireDate(statement)) {
-                model.add(toDctermsDate(statement));
-            } else if (isHighWireLanguage(statement)) {
-                model.add(toDctermsLanguage(statement));
-            } else if (isHighWireAuthor(statement)) {
-                model.add(toDctermsCreator(statement));
-            } else {
-                model.add(statement);
-            }
+            extractKnownProperties(model, statement);
         }
         return model;
     }
 
-    private Statement toDctermsCreator(Statement statement) {
-        return valueFactory.createStatement(statement.getSubject(),
-                DcTerms.CREATOR.getIri(valueFactory), statement.getObject());
+    private void extractKnownProperties(Model model, Statement statement) throws IOException, InterruptedException {
+        OntologyProperty ontologyProperty = getMappedOntologyProperty(statement);
+        Value value = extractValue(ontologyProperty, statement.getObject());
+        OntologyProperty mappedProperty = mapToSpecificProperty(ontologyProperty, value);
+        if (nonNull(ontologyProperty)) {
+            model.add(statement.getSubject(), mappedProperty.getIri(valueFactory), value);
+        }
     }
 
-    private boolean isHighWireAuthor(Statement statement) {
-        return statement.getPredicate().toString().endsWith(CITATION_AUTHOR);
+    private OntologyProperty mapToSpecificProperty(OntologyProperty ontologyProperty, Value value) {
+        return value.toString().startsWith("https://doi.org/") ? Bibo.DOI : ontologyProperty;
     }
 
-    private Optional<Statement> toBiboDoi(Statement statement) throws IOException,
+    private Value extractValue(OntologyProperty ontologyProperty, Value object) throws IOException,
             InterruptedException {
-        String value = statement.getObject().stringValue();
-        return extractDoi(value).map(doiString -> valueFactory.createStatement(statement.getSubject(),
-                valueFactory.createIRI(BIBO_DOI), valueFactory.createLiteral(doiString)));
+        if (Bibo.DOI.equals(ontologyProperty) || DcTerms.IDENTIFIER.equals(ontologyProperty) && isDoiString(object)) {
+            return extractDoi(object.stringValue());
+        } else {
+            return object;
+        }
     }
 
-    private Optional<String> extractDoi(String value) throws IOException, InterruptedException {
-        return isShortDoi(value) ? fetchDoiUriFromShortDoi(value)
+    private OntologyProperty getMappedOntologyProperty(Statement statement) {
+        String property = statement.getPredicate().getLocalName();
```

Comment:
For now, this works, but it will not when we add further namespaces. This is not in-scope at present